lib/jwt/jwt.go

```diff
@@ -265,6 +266,65 @@ func (k *Key) SignAWSOIDC(p SignParams) (string, error) {
 	return k.sign(claims, opts)
 }
 
+// SignParamsJWTSVID are the parameters needed to sign a JWT SVID token.
+type SignParamsJWTSVID struct {
+	// JTI is the unique JWT ID.
+	JTI string
+	// SPIFFEID is the SPIFFE ID of the workload to which it is issued.
+	SPIFFEID spiffeid.ID
+	// Audiences are the audiences to include in the token as the expected
+	// recipients of the token.
+	Audiences []string
+	// TTL is the time to live for the token.
+	TTL time.Duration
+}
+
+// SignJWTSVID signs a JWT SVID token.
+// See https://github.com/spiffe/spiffe/blob/main/standards/JWT-SVID.md
+func (k *Key) SignJWTSVID(p SignParamsJWTSVID) (string, error) {
+	claims := jwt.Claims{
+		// > 3.1. Subject:
+		// > The sub claim MUST be set to the SPIFFE ID of the workload to which it is issued.
+		Subject: p.SPIFFEID.String(),
+		// > 3.2. Audience:
+		// > The aud claim MUST be present, containing one or more values.
+		Audience: p.Audiences,
+		// > 3.3. Expiration Time:
+		// > The exp claim MUST be set
+		Expiry: jwt.NewNumericDate(k.config.Clock.Now().Add(p.TTL)),
+		// The spec makes no comment on inclusion of `iat`, but the SPIRE
+		// implementation does set this value and it feels like a good idea.
+		IssuedAt: jwt.NewNumericDate(k.config.Clock.Now()),
```

Comment:
nit for cleanliness: get the current time once so the difference between exp and iat is exactly the TTL rather than potentially TTL minus one second

Reply:
b442470a84